js/vnext/projection/jsdata.js

```diff
@@ -0,0 +1,41 @@
+import _ from 'underscore';
+import Promise from 'bluebird';
+
+export function jsdata (state, {
+  query,
+  entity,
+  options,
+  skip,
+  take,
+  filter,
+  orderby = [],
+  select = [],
+} = {}) {
+  const op = {};
+
+  if (take) {
+    op.limit = take;
+  }
+
+  if (skip) {
+    op.offset = skip;
+  }
+
+  if (filter) {
+    op.where = filter;
+  }
+
+  if (query) {
+    op.query = query;
+  }
+
+  if (orderby && orderby.length) {
+    op.orderby = _.reduce(orderby, (memo, obj) => {
+      _.each(obj, (key, value) => {memo.push([key, value > 0 ? 'ASC' : 'DESC']);});
+      return memo;
+    }, []);
+  }
+
+  return entity.fincAll(op, _.defaults(options, { all: true }))
+  .then(_.property('value'));
```

Comment:
sublime auto reindent does not work on this file~ So I fix the indentation by hand T_T